js/interactive-guides/externalized-string.js

```diff
@@ -0,0 +1,182 @@
+/******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ ******************************************************************************/
+var supportedLanguages = ["cs",
+                          "de",
+                          "en",
+                          "es", 
+                          "fr",
+                          "hu",
+                          "it",
+                          "ja",
+                          "ko",
+                          "pl",
+                          "pt-br",
+                          "ro",
+                          "ru",
+                          "zh",
+                          "zh-tw"
+                          ];
+
+/**
+ * Check if browser language is supported by us
+ */
+function checkIfLanguageSupported(browserLanguage){
+    "use strict";
+    var languageCode = "";
+    
+    if (supportedLanguages.indexOf(browserLanguage) !== -1) {
+        languageCode = browserLanguage;
+    } else if (browserLanguage.length === 5) { 
+        var code = browserLanguage.substring(0, 2);
+        if(supportedLanguages.indexOf(code) !== -1) {
+            languageCode = code;
+        }
+    }  
+  
+    return languageCode;
+}
+
+/**
+ * Get language code from the browser's list of languages
+ */
+function getLanguageCode(){
+    "use strict";
+    
+    // Obtain browser languages
+    var browserLanguages = navigator.languages;
+    var languageCode = "";
+    var browserLanguage = "";
+    
+    // Loop through languages and check if supported 
+    if (browserLanguages) {
+        for(var i=0; i<browserLanguages.length; i++){
+            languageCode = checkIfLanguageSupported(browserLanguages[i].toLowerCase());
+            if(languageCode){
+                break;
+            }
+        }
+    } else {
+        browserLanguage = navigator.language ? navigator.language : navigator.userLanguage;
+        languageCode = checkIfLanguageSupported(browserLanguage.toLowerCase());
+    }
+    // Default to English if no language found
+    if (languageCode === ""){
+        languageCode = "en";
+    }
+
+    return languageCode;
+}
+
+/**
+ * Query all the data-externalizedString in html to translate the message.
+ */
+function retrieveExternalizedStrings(rootDir) {
+    'use strict';
+        
+    var languageCode = getLanguageCode();
+
+    var nlsFile = "messages.js";
+    var url = (languageCode !== "en") ? rootDir + languageCode + "/" + nlsFile :  rootDir + nlsFile;
```

Comment:
This is assuming `rootDir` ends with a `/`. Maybe note it in comment or have a check for it.